package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/repository/impl/JdbcAssessmentPackageRepository.java

```
@@ -86,6 +104,110 @@ public void createPackage(final Collection<Assessment> assessments, final long i
         if (!assessmentNaturalIds.isEmpty()) jdbcTemplate.update(sqlUpdateDifficultyCode, new MapSqlParameterSource("asmt_natural_ids", assessmentNaturalIds));
     }
 
+    @Transactional(readOnly = true)
+    @Override
+    public Assessment findOneByNaturalId(final String id) {
+
+        final List<Assessment.Builder> assessmentList = jdbcTemplate.query(slqFindOneByNaturalId, new MapSqlParameterSource("natural_id", id), new AssessmentRowMapper());
+        if (assessmentList.size() == 1) {
```

should it throw if it gets more than 1 instead of returning null?  if there are 0 then of course null, but if there are 2?

---

Well, I guess I know that it is UNIQUE in the DB...And if it is not unique,  a lot of things would go wrong. But then from this code point of view I can check for more than 1 and throw an exception. Will do.

---

I waffled on this one too. I think it should stay as is. There is NO way to generate the error that you are checking for (without changing the schema).

---

For some reason GitHub was hiding this from me.  Since the DB has a unique constraint and we aren't worried about having 2, shouldn't the check be `if (!assessmentList.isEmpty())` (or check for empty and return null first).  Since you don't really care about the size of the list just that something exists.  It just feels odd to check for exactly 1 when if it's 2 not doing anything about it.  So seems like an existence check makes more since and is microscopically faster :)

Not a big deal though.

---

Makes sense. Will adjust with the next change.